src/Poseidon/Package.hs

```diff
@@ -280,3 +330,62 @@ compFunc1 (EigenstratSnpEntry c1 p1 _ _ _ _, _) (EigenstratSnpEntry c2 p2 _ _ _
 compFunc2 :: (EigenstratSnpEntry, GenoLine) -> [(EigenstratSnpEntry, GenoLine)] -> Ordering
 compFunc2 (EigenstratSnpEntry c1 p1 _ _ _ _, _) ((EigenstratSnpEntry c2 p2 _ _ _ _, _):_) = compare (c1, p1) (c2, p2)
 compFunc2 _                                     []                                        = error "compFunc2 - should never happen"
+
+-- Janno file loading
+
+decodingOptions :: Csv.DecodeOptions
+decodingOptions = Csv.defaultDecodeOptions { 
+    Csv.decDelimiter = fromIntegral (ord '\t')
+}
+
+instance Csv.FromRecord PoseidonSample
+
+-- | A helper function for parsing double lists
+bytestringToDouble :: [Bch.ByteString] -> [Double]
+bytestringToDouble [] = []
+bytestringToDouble (x:xs) = (read (Bch.unpack x) :: Double) : bytestringToDouble xs
```

Comment:
OK, using `read` is generally considered unsafe because this will fail ungracefully if the unpacked string can't be read as an integer. Have you checked what happens in such a case?